### DISCUSSION TITLE
Preventing text wrapping inside tables inside grid environment

### DISCUSSION THREAD

**Discussion by psads-git** [typst/typst#4507] [Q&A] [answered]
I am using the code below, which shows a way of trying to prevent text wrapping in the tables in the grid (when I use `tbl1`), but success. Could you please help me? Thanks!

```
#set enum(full: true, numbering: (..n) => {
    let format = if n.pos().len() > 1 {"a)"} else {"1."}
    numbering(format, n.pos().last())
})

#let Q01 = [
Consider the following matrix:

$
mat(
  1, 2;
  4, 9;
)
$]

#let Q02 = [
Solve the following integral:

$
integral x e^(x) dif x.
$]

#let l = (Q02, Q01)

#let tbl = [#table(
  columns: 3,
  table.header(
    [Substance],
    [Subcritical °C],
    [Supercritical °C],
  ),
  [Hydrochloric Acid],
  [12.0], [92.1],
  [Sodium Myreth Sulfate],
  [16.6], [104],
  [Potassium Hydroxide],
)]

#tbl
#let tbl2 = scale(x:50%, y: 50%,  [ #tbl])
#tbl2

#for i in range(l.len()) {
  [+ #l.at(i)
        #block(
          breakable: false,
          grid(
            columns: 2,
            rows: 2,
            gutter: 0.5em,
            enum.item(1)[#tbl2],
            enum.item(2)[#tbl],
            enum.item(3)[#scale(x:80%, y: 80%,  [ #tbl])],
            enum.item(4)[#tbl])
        )
  ]
}

```

![image](https://github.com/typst/typst/assets/75945439/89cedf7d-8a4c-4aa5-b62b-885cf3eb9f82)

**Comment by Mc-Zen** [comment]
Hi, the default column width is `auto` which means that it tries to make the column as wide as its content. But if there is not enough space available (which is the case for the grid at the bottom), the columns stop growing. You can specify the column width manually by passing an array of `length`, `ratio`, and `relative length` values instead of `3` to `columns`. The fact is that it still won't fit here (as I just tried) because it is just too wide. 

Note that `scale` does not affect the layout, so the scaled table does not "know" that it would have "more space" now.

**Comment by TS60** [comment]
Probably, because you have not enough space left in the second example.
Maybe try to give your page more space `#set page(margin: (left: 1mm, right: 1mm))` instead of experimenting with scale.

**Comment by psads-git** [comment]
Thanks, @Mc-Zen and @TS60, for your replies.

In LaTeX, there is the package `adjustbox`, which allows us to scale a table completely preserving the appearance of the original table:

[`adjustbox`](https://www.ctan.org/pkg/adjustbox)

I suspect that, internally, `adjustbox` creates an image of the original table and then scales it. Doing so, the appearance of the table does not change with scaling.

**Comment by Mc-Zen** [comment, marked as answer]
Ah, yes if that's want you want to achieve, then no problem. Just use this version instead of `scale`:

```typ
#let mscale(factor, content) = context {
  let size = measure(content)
  return scale(factor, reflow: true, block(..size, content))
}
```
First, we measure the table. Note, that in this setting, `measure` does not know how much space is available and assumes infinite space. We then force our table into a block of this size and scale it. `reflow: true` guarantees, that Typst "sees" the table in the new size, and not the original size. 

Let me know if this solved your request!

**Comment by psads-git** [comment]
Thanks a lot, @Mc-Zen! That has fixed the issue!
